lib/rspec/mocks/error_generator.rb

```diff
@@ -288,11 +287,12 @@ def arg_has_valid_description?(arg)
       end
 
       def format_received_args(*args)
-        args.empty? ? "(no args)" : "(" + received_arg_list(*args) + ")"
-      end
-
-      def received_arg_list(*args)
-        args.map(&:inspect).join(", ")
+        return "(no args)" if args.empty? || args.first.empty?
+        return formatted_single_message(args) if single_message?(args)
+        grouped_list = grouped_args(args).map do |arg, index|
+          [formatted_single_message([arg]), group_count(index, args)].join
+        end
+        grouped_list.join("\n            ")
       end
```

Comment:
Maybe I'm missing something, but it looks like this method can be simplified to:

``` ruby

def format_received_args(*args_for_multiple_calls)
  grouped_args(args_for_multiple_calls).map do |args_for_one_call, index|
    [format_args(*args_for_one_call), group_count(index, args_for_multiple_calls)].join
  end.join("\n            ")
end
```

(Notice also the renaming of the arguments, which I think are more clear.  You're version also has `arg`, which is actually a list of args and not a single arg, right?)